Documentation/Ko-kr/ArrangeActAssert.md

```diff
@@ -0,0 +1,203 @@
+# XCTest를 사용한 효과적인 테스트 : Arrange, Act, 그리고 Assert
+
+XCTest, Quick, 또는 다른 테스트 프레임워크를 사용하는 경우, 간단한 패턴을 따라 효과적인 단위 테스트를 작성할 수 있습니다:
+
+1. Arrange (환경 구축)
+2. Act (실행)
+3. Assert (동작 확인)
+
+## Arrange, Act, and Assert 사용하기
+
+예를 들어, `Banana`라는 클래스를 간단히 살펴봅시다:
+
+```swift
+// Banana/Banana.swift
+
+/** A delicious banana. Tastes better if you peel it first. */
+public class Banana {
+  private var isPeeled = false
+
+  /** Peels the banana. */
+  public func peel() {
+    isPeeled = true
+  }
+
+  /** You shouldn't eat a banana unless it's been peeled. */
+  public var isEdible: Bool {
+    return isPeeled
+  }
+}
+```
+
+ `Banana.peel()`  메서드가 어떻게 되는지 확인해 봅시다:
+
+```swift
+// BananaTests/BananaTests.swift
+
+class BananaTests: XCTestCase {
+  func testPeel() {
+    // Arrange: Create the banana we'll be peeling.
+    let banana = Banana()
+
+    // Act: Peel the banana.
+    banana.peel()
+
+    // Assert: Verify that the banana is now edible.
+    XCTAssertTrue(banana.isEdible)
+  }
+}
+```
+
+## 간단한 테스트 이름 사용하기
+
+이 `testPeel()` 테스트 덕분에 `Banana.peel()` 메소드가 제대로 작동하지 않는다면 즉시 알 수 있습니다. 이는 일반적으로 응용 프로그램 코드가 변경되면서 발생합니다.
+
+1. 우리는 뜻하지 않게 응용 프로그램 코드를 망가트리므로, 응용 프로그램 코드를 수정해야 합니다.
+2. 우리는 응용 프로그램의 작동 방식을 변경했습니다. — 아마도 새로운 기능을 추가했기 때문일 수 있습니다. — 그래서 우리는 테스트 코드를 변경해야 합니다. 
+
+만약 테스트가 실패하기 시작한다면, 위 두 가지 중 어떤 케이스인지 어떻게 알 수 있을까요? 따라서 **테스트의 이름**을 알기 쉬운 것이 중요하다는 사실을 알려줍니다. 좋은 테스트 이름들 :
+
+1. 무엇을 테스트하고 있는지 명확합니다.
+2. 어떤 때 테스트가 통과하거나 실패하는지 명확합니다.
+
+ `testPeel()` 메소드는 좋은 테스트 이름일까요? 더 명확하게 바꾸어 봅시다:
+
+```diff
+// BananaTests.swift
+
+-func testPeel() {
++func testPeel_makesTheBananaEdible() {
+  // Arrange: Create the banana we'll be peeling.
+  let banana = Banana()
+
+  // Act: Peel the banana.
+  banana.peel()
+
+  // Assert: Verify that the banana is now edible.
+  XCTAssertTrue(banana.isEdible)
+}
+```
+
+새로운 이름:
+
+1. 무엇을 테스트하고 있는지 명확합니다 : `testPeel` 은  `Banana.peel()` 메소드 임을 나타냅니다.
+2. 어떤 때 테스트가 통과하거나 실패하는지 명확합니다 : `makesTheBananaEdible` 는 메소드가 호출되면 바나나가 식용 가능한 상태임을 나타냅니다.
+
+## 테스트 시 조건들
+
+`offer()` 함수를 사용해서 사람들에게 바나나를 제공하고 싶다고 가정해 보겠습니다 :
+
+```swift
```

Comment:
In the other languages, this is \`\`\`diff, not \`\`\`swift, which makes GitHub highlight the added/removed lines a little better.